psi4/src/psi4/libmints/molecule.cc

```diff
@@ -1287,7 +1297,8 @@ std::shared_ptr <Molecule> Molecule::create_molecule_from_string(const std::stri
         }
     }
 
-    mol->set_has_zmatrix(zmatrix);
+    mol->set_has_zmatrix(has_zmatrix);
+    mol->set_has_cartesian(has_cart);
```

Comment:
Mixed will have both and we freeze the COM/Orient on detection of any cart. Its why we needed the extra cart detection.